src/Microsoft.Extensions.WebEncoders.Core/UrlPathDecoder.cs

```diff
@@ -0,0 +1,326 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.Extensions.WebEncoders
+{
+    /// <summary>
+    /// UrlPathDecoder unescape a given path string.
+    /// </summary>
+    public class UrlPathDecoder
+    {
+        private static Lazy<Encoding> _utf8 = new Lazy<Encoding>(() =>
+        {
+            return Encoding.GetEncoding(
+                "utf-8",
+                new EncoderExceptionFallback(),
+                new DecoderExceptionFallback());
+        }, isThreadSafe: true);
```

Comment:
Never seen the `isThreadSafe` parameter used before. Apparently `true` is already the default so this isn't necessary: https://msdn.microsoft.com/en-us/library/dd997286(v=vs.110).aspx